src/libraries/Microsoft.PowerFx.Core/Public/Types/AggregateType.cs

```diff
@@ -46,6 +46,39 @@ public virtual bool TryGetFieldType(string name, out FormulaType type)
             return true;
         }
 
+        /// <summary>
+        /// Takes display or logical name as input and returns field and logical name if exists.
+        /// </summary>
+        /// <param name="displayOrLogicalName">Display or Logical name.</param>
+        /// <param name="logical">Logical name for the input.</param>
+        /// <param name="type">Type for the input Display or Logical name.</param>
+        /// <returns>true or false.</returns>
+        /// <exception cref="ArgumentNullException">Throws, if input displayOrLogicalName is empty.</exception>
+        public virtual bool TryGetFieldType(string displayOrLogicalName, out string logical, out FormulaType type)
```

Comment:
IMO, this shouldn't be virtual. It gets the logical name, and then calls `TryGetFieldType`, which is virtual. Not sure why we'd want to allow overriding this.

Reply:
+1 
The case to consider here is a derived RecordType that overrides FieldNames. 

TryGetFieldType is already virtual. 
And the display-->logical mapping is via the DisplayNameProvider, which also has virtuals. 

The piece that's missing though is how would the derived RecordType provide it sown DisplayNameProvider? 

Maybe we need to allow that as part of deriving.

Reply:
PR for how would the derived RecordType provide it sown DisplayNameProvider?
https://github.com/microsoft/Power-Fx/pull/593